### ISSUE TITLE
ec2-instance-connect open-tunnel failed when NO_PROXY not set.

### ISSUE THREAD

**Issue by sayurin** [aws/aws-cli#8023]
### Describe the bug

ec2-instance-connect open-tunnel failed when NO_PROXY not set.

### Expected Behavior

open-tunnel successfully connected.

### Current Behavior

following error message.

```
2023-07-07 17:47:08,068 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
Traceback (most recent call last):
  File "awscli\clidriver.py", line 460, in main
  File "awscli\clidriver.py", line 595, in __call__
  File "awscli\customizations\commands.py", line 205, in __call__
  File "awscli\customizations\ec2instanceconnect\opentunnel.py", line 145, in _run_main
  File "awscli\customizations\ec2instanceconnect\websocket.py", line 326, in run
  File "awscli\customizations\ec2instanceconnect\websocket.py", line 363, in _open_websocket_connection
  File "awscli\customizations\ec2instanceconnect\websocket.py", line 158, in connect
TypeError: argument of type 'NoneType' is not iterable

argument of type 'NoneType' is not iterable
kex_exchange_identification: Connection closed by remote host
```

### Reproduction Steps

1. Setup EC2 Instance, EC2 Instance Connect and EC2 Instance Connect Endpoint. (Instance ID as i-xxxx).
2. Setup HTTP Proxy (Proxy address as proxy.example.com:8080).
3. Set environment variable `HTTP_PROXY` to use proxy.
    In bash, `export HTTP_PROXY=http://proxy.example.com:8080`.
    In PowerShell `$env:HTTP_PROXY="http://proxy.example.com:8080"`.
4. Unset env `NO_PROXY`.
    In bash, `unset NO_PROXY`.
    In PowerShell, `$env:NO_PROXY=""`. 
5. Run aws cli `aws ec2-instance-connect ssh --connection-type eice --instance-id i-xxxx --debug`.
    This will fail with the error written at Current Behavior.

Workarround,

1. Set environment variable `NO_PROXY` to dummy value.
    In bash, `export NO_PROXY=1`.
    In PowerShell, `$env:NO_PROXY="1"`.
2. Run aws cli `aws ec2-instance-connect ssh --connection-type eice --instance-id i-xxxx --debug`.
    This will success as expected behavior.

### Possible Solution

https://github.com/aws/aws-cli/blob/03e30287bf02f7d47bafdcbf4aaca0043f7cec11/awscli/customizations/ec2instanceconnect/websocket.py#L157-L158

If environment variable `NO_PROXY` is not set, `no_proxy` become `NoneType`.

### Additional Information/Context

For workaround, set environment variable `NO_PROXY` with dummy value.

### CLI version used

4.12.7

### Environment details (OS name and version, etc.)

Windows and CloudShell.

**Comment by tim-finnigan** [comment]
Hi @sayurin thanks for reaching out. I'm having trouble reproducing this issue, could you share your debug logs (with any sensitive info redacted) by adding `--debug` to your command?

**Comment by sayurin** [comment]
@tim-finnigan I updated description, repro steps. Please try with `--connection-type eice` option.

**Comment by tim-finnigan** [comment]
Closing as https://github.com/aws/aws-cli/pull/8039 was merged, please update to v2.13.2+ per the [CHANGELOG](https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst?plain=1) and let us know if you run into any issues. Thanks again for reporting this!
